homeassistant/components/withings/__init__.py

```diff
@@ -87,8 +81,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     config_updates = {}
 
     # Add a unique id if it's an older config entry.
-    if entry.unique_id != entry.data["token"]["userid"]:
-        config_updates["unique_id"] = entry.data["token"]["userid"]
+    if entry.unique_id != entry.data["token"]["userid"] or not isinstance(
+        entry.unique_id, str
+    ):
+        config_updates["unique_id"] = str(entry.data["token"]["userid"])
 
     # Add the webhook configuration.
     if CONF_WEBHOOK_ID not in entry.data:
```

Comment:
Could you be a bit more specific? I'm not sure what to fix.

Reply:
The downside of a migration of the config entry is that it makes it impossible to roll back to a  earlier home assistant version.

Reply:
@MartinHjelmare Basically, the same thing is happening here. The schema is changed/migrated... ?

Reply:
In this case we don't seem to change the structure of the existing entry data, so here it doesn't matter what migration approach we take I think.

What I meant was that if the existing data is moved to a different structure in the entry data, a rollback is not possible. As long as we're only adding items to the root level of the dict and don't touch the existing root level items, a rollback is possible.

Reply:
The code was changed because `ConfigFlow.async_set_unique_id()` expects a string for the `unique_id` and I figured this should be updated to stay consistent with the rest of the HASS code base.
Should this be changed/removed?

Reply:
I think it is fine to leave as is right now. I just noticed it and is out of scope for this PR I would say. 👍 

Let's resolve this conversation for now and focus on the actually PR 😉